lib/src/data/tutor_availability.dart

```diff
@@ -0,0 +1,58 @@
+import "utils.dart";
+
+extension type TutorID(String value) {}
+
+class TutorAvailability {
+  final TutorID tutorId;
+  final String department;
+  final List<String> availableCourses;
```

Comment:
```suggestion
  final List<CourseID> availableCourses;
```

And define a CourseID in the meantime until that branch is merged